pdns/dnsdist-tcp.cc

```diff
@@ -190,7 +190,7 @@ static bool sendResponseToClient(int fd, const char* response, uint16_t response
 static bool maxConnectionDurationReached(unsigned int maxConnectionDuration, time_t start, unsigned int& remainingTime)
 {
   if (maxConnectionDuration) {
-    time_t elapsed = time(NULL) - start;
+    unsigned int elapsed = time(NULL) - start;
```

Comment:
It probably doesn't matter much but this would lead to closing the connection right away if the time goes backward, because then `elapsed` becomes huge instead of negative. Perhaps we should store `time(nullptr)`, and if it's lesser than `start` set `elapsed` to 0.